### PR TITLE
Added check for command textbox

### PR DESCRIPTION
**Pull request by nbhoski**
Fixed geck # 2221893 

* `Command` text box under `Run MATLAB Command` now checks if the the command box is empty 
* Empty command box shows an error message 

image

* Added Integ tests

### CODE REVIEW

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java [hunk 1]
```diff
@@ -110,6 +114,14 @@ public boolean isApplicable(
...
             return true;
         }
+        @POST
+        public FormValidation doCheckMatlabCommand(@QueryParameter String matlabCommand) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+            if (matlabCommand.isEmpty()) {
+                return FormValidation.error(Message.getValue("matlab.empty.command.error"));
+            }
+            return FormValidation.ok();
+        }
```
Comment (sameagen-MW) [inline src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:124]:
```suggestion
        public FormValidation doCheckMatlabCommand(@QueryParameter String value) {
            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
            if (value.isEmpty()) {
                return FormValidation.error(Message.getValue("matlab.empty.command.error"));
            }
            return FormValidation.ok();
        }

```

Comment (sameagen-MW) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:124]: I believe the parameter name is significant here, and needs to be `value`. See [this](https://pitchfork.com/reviews/albums/britney-spears-blackout/) page for reference.

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java [hunk 2]
```diff
@@ -110,6 +114,14 @@ public boolean isApplicable(
                 @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
             return true;
         }
+        @POST
```
Comment (sameagen-MW) [inline src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:117]: I don't think that this is necessary. Also, we should add a blank line above it for whitespace consistentcy.

File: src/main/resources/com/mathworks/ci/freestyle/RunMatlabCommandBuilder/config.jelly [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     
 	  <f:entry title="Command" field="matlabCommand">
-	        <f:textbox/>
+	        <f:textbox checkMethod="post"/>
```
Comment (sameagen-MW) [inline src/main/resources/com/mathworks/ci/freestyle/RunMatlabCommandBuilder/config.jelly:5]: I don't think that this should be necessary. See [here](https://pitchfork.com/reviews/albums/britney-spears-blackout/).

File: src/test/java/integ/com/mathworks/ci/RunMatlabCommandBuilderTest.java [hunk 4]
```diff
@@ -7,6 +7,9 @@
  * 
  */
 
+import com.gargoylesoftware.htmlunit.WebAssert;
+import com.gargoylesoftware.htmlunit.html.HtmlCheckBoxInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
```
Comment (sameagen-MW) [inline src/test/java/integ/com/mathworks/ci/RunMatlabCommandBuilderTest.java:12]: I see that the pom file hasn't been updated. Is this already a Jenkins dependency?

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java [hunk 5]
```diff
@@ -110,6 +114,14 @@ public boolean isApplicable(
                 @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
             return true;
         }
+        @POST
+        public FormValidation doCheckMatlabCommand(@QueryParameter String matlabCommand) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```
Comment (sameagen-MW) [inline src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:119]: Is there any reason why we're checking this? I would expect that Jenkins should already handle the permissions of who can edit a pipeline without us needing to do that.

**Review by sameagen-MW — CHANGES_REQUESTED** [review]
This looks slick. The integration tests especially look great. There's just a couple of things that I'm not sure are totally necessary or at least that I don't fully understand at the moment.

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java (hunk 1 above)
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:124]: Do you mean [this](https://www.jenkins.io/doc/developer/security/form-validation/) link

Comment (nbhoski) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:124]: I had named it to be in synch with our `UseBuildWrapper` class validations https://github.com/mathworks/jenkins-matlab-plugin/blob/a66bd3a3262d42937074bb783871456986c6d2b5/src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java#L145

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java (hunk 2 above)
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:117]: This Recommended approach for avoiding CSRF attacks please refer [this](https://www.jenkins.io/doc/developer/security/form-validation/#protecting-from-csrf)  also we had received the Jenkins issue from Jenkins maintainers earlier to use this approach from security perspective

File: src/main/resources/com/mathworks/ci/freestyle/RunMatlabCommandBuilder/config.jelly (hunk 3 above)
Comment (nbhoski) [reply src/main/resources/com/mathworks/ci/freestyle/RunMatlabCommandBuilder/config.jelly:5]: same as above the link is broken i have referred [this](https://www.jenkins.io/doc/developer/security/form-validation/#protecting-from-csrf)

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java (hunk 5 above)
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:119]: Same as above the read permission is implied hence such static methods are susceptible for security issues.

File: src/test/java/integ/com/mathworks/ci/RunMatlabCommandBuilderTest.java (hunk 4 above)
Comment (nbhoski) [reply src/test/java/integ/com/mathworks/ci/RunMatlabCommandBuilderTest.java:12]: Yes , We already have HTML unit tests and dependencies are part of it.

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java (hunk 1 above)
Comment (sameagen-MW) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java:125]: Haha, sorry about that link. I think I was talking about music with a friend right about the same time I was writing the review. [This](https://www.jenkins.io/doc/developer/forms/form-validation/) was the link that I meant to send. 

 I think that you're right that there's no downside to taking the more secure option here, so you can just ignore this comment chain.

**Review by sameagen-MW — CHANGES_REQUESTED** [review]
Looks good, I think the only thing I see is that we probably need to either change the validation function parameter name to `value` or use `@QueryParameter("value")`, as outlined [here](https://www.jenkins.io/doc/developer/forms/form-validation/#form-validation).

File: src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java [hunk 6]
```diff
@@ -110,6 +114,15 @@ public boolean isApplicable(
                 @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
             return true;
         }
+
+        @POST
+        public FormValidation doCheckMatlabCommand(@QueryParameter String matlabCommand) {
```
[the remark this answers is not in the document]
Comment (nbhoski) [reply src/main/java/com/mathworks/ci/freestyle/RunMatlabCommandBuilder.java]: Done

**Review by sameagen-MW — APPROVED** [review]

**Pull request merged**